### ISSUE TITLE
Re-assessing sneak jump / sneak ladder simulations in new movement code

### ISSUE THREAD

**Issue by paramat** [minetest/minetest#7107]
New movement code was added to fix damage-avoiding bugs, but this changed sneak behaviour, so we then added non-identical simulations of sneak jump and sneak ladder to it, and later added an option for using the old movement code. The old movement code was initially meant to be temporary but now due to huge player and server demand it is now going to be kept.

My suspicion is that those who like the old sneak jump and sneak ladder behaviours use the option for old movement code despite the damage evasion bug, and do not like the non-identical simulations in new movement code, i would like feedback on this.
It was difficult work to add these simulations and they seem to compromise how well new movement code and new standard sneak behaviour works. They were only added while we were intending to eventually remove old movement code.

My suggestion is to consider removing these simulations of sneak jump and sneak ladder from new movement code, to possibly allow new movement behaviour and new standard sneak behaviour to be improved, and also to simplify new movement code.
If done we of course will have to commit to keeping the option for old movement code, we already do commit to that but this is worth stating to be clear.

@SmallJoker you are the movement code specialist and i seem to remember you might be interested in further work on improving movement code, what are your thoughts on this?

**Comment by paramat** [comment]
I think that simulations of the old sneak side-effect behaviours have no place in new movement code now that we are keeping the option for old movement code. People who prefer the old movement are obviously happy with it and will accept the damage evasion bug that unavoidably goes with it.

The simulation of sneak jump is actually very different as it only allows a 2 node jump, instead of boosting what ever jump height the player already has. I know that Ezhh finds the simulated sneak jump unusable because of this, understandably. I expect many players feel the same.

Servers that were built to work with the old movement code require the exact old behaviour and i expect have no use for the new simulations.

**Comment by 4w** [comment]
> i would like feedback on this.

Remove the bug code and bug simulation code without replacement. Everyone who wants to have it back should use a mod for that.

**Comment by Ezhh** [comment]
New move, even with the current changes, has big problems for me - the 2 node jump issue, plus the change in controls  (especially the change in controls!) means I'm never going to want to use it, so as long as old move code is kept, I don't mind what happens to new. 

One thing you might want to get some opinions on is whether those using new movement on some servers also want to preserve sneak ladder, since for some people sneak ladder seemed to be the only issue with the change. (Note: I'd hate to see another change to sneak that upset people, regardless of whether it has any effect on me or my servers.)

**Comment by 4w** [comment]
> for some people sneak ladder seemed to be the only issue with the change.

Of course people like you are butthurt about fixed bugs (hence the downvote I guess). This does not make bugs better in any way. Just don't rely on bugs for your functionality.

**Comment by Ezhh** [comment]
Downvote because I know what it would cause if old move code was removed, and I actually care about not having the community go through another big argument. 

Second downvote because basic politeness seems an alien concept to you.  Some of us can manage to disagree without needing insults. :)

**Comment by 4w** [comment]
> Second downvote because

You don't like me because I am not a "nice guy". I know, but actually I couldn't care less.

> basic politeness seems an alien concept to you.

Politeness is irrelevant in software development. Be honest and say what you think. Be technically correct, avoid making lame compromises, fight bugs, always give 100%, bugs are not features.

Enough with off-topic now.

**Comment by rubenwardy** [comment]
Whilst I like fixing bugs, this bug has lead to well loved behaviour. Old move should be kept, and is not the point of this issue.

I think that new move could be improved to be more fun without adding bugs. However, if this does not happen, no one uses the settings in new move, and it adds maintenance cost then it should be removed.

> Politeness is irrelevant in software development. 

Bullshit. Basic decency is relevant in any human interaction.

**Comment by 4w** [comment]
> Whilst I like fixing bugs, this bug has lead to well loved behaviour.

A bug is not a feature and should not be seen as one. If constant bug using leads to a certain behavior by a lot of users then the bug should be fixed and the effects of the bug should be properly re-implemented as optional (-> maximum configurability) feature.

In no way the buggy code should be re-used.

> Basic decency is relevant in any human interaction.

Decency != politeness

**Comment by Ezhh** [comment]
> ..should be properly re-implemented as optional (-> maximum configurability) feature.

This is what was asked for. 

Despite your silly assumptions I don't care about the "bug" part of old move code which was the main problem with it (the damage avoidance when falling) and would have been happy enough to see that cheat removed, but the process of removing that also changed other behaviour that was wanted and had nothing to do with cheating. For example, the controls for sneak were changed in new movement code in a way that I, and other users, found made character movement less precise (before you could release shift to immediately drop, while now you release then move to drop). I vaguely recall it being said this was to make it easier for app users (someone please correct me if wrong), but not all players are app users.

> In no way the buggy code should be re-used.

This issue isn't about whether buggy code should be reused. It's about whether new movement should continue to attempt to emulate certain (non-buggy, but preferred by some people) features of old movement, when people who prefer old movement will stick with that anyway.

**Comment by benrob0329** [comment]
I say make the new movement code better, in new ways. People can use the old code for now if they like the classic behavior, but let's make a new set of beloved movement styles.

Don't emulate tradition, just make a better one.

**Comment by 4w** [comment]
> the controls for sneak were changed

So it isn't Shift anymore and it doesn't prevent falling from node edges anymore?

> (before you could release shift to immediately drop, while now you release then move to drop)

This is intentional if I remember correctly.

**Comment by Ezhh** [comment]
Intentional yes, but you can't tell me the original "release shift to fall" was a "bug".

**Comment by SmallJoker** [comment]
Before this topic gets too large, I'd like to say that there's no pressure on this issue at all. These 300 lines of code don't hurt anybody and the localplayer code is in an acceptable size.

Changing the new sneak code is totally possible but are there any explicit suggestions? @benrob0329
> I say make the new movement code better, in new ways.

or @Ezhh
>  has big problems for me - the 2 node jump issue, plus the change in controls (especially the change in controls!)

Regarding the first one: These jumps are still possible when the sneak glitch is enabled. What exactly is the issue with it? If you need an additional boost, then it can be implemented by using the X/Z velocity which athletes use to jump higher. Free boost while sneaking is strange and should be kept in the old move code.
For the latter it's also not clear to me what you mean with the "change in controls". Could you please elaborate on these changes?

----
What also might be interesting are a few poll results that show what the players actually use, so this issue can be resolved easier.

**Comment by Ezhh** [comment]
@SmallJoker

>  (before you could release shift to immediately drop, while now you release then move to drop

Needing that additional action takes away a lot of control/accuracy. Holding shift, for me, equated with holding on to the edge; releasing with letting go, and therefore falling.

To the moaners (I'm looking at you "not a nice guy" 4w!), I'm not in anyway claiming the old controls are the only right/correct approach, but the old controls were certainly not a bug and I much prefer them.

> These jumps are still possible when the sneak glitch is enabled. What exactly is the issue with it?

The 2 node jumping issue is that players can equip items on many servers (including mine) which enable 2 node high jumping by default. Logically the sneak jump should then allow for a 3 node jump. Sneak jump should be base jumping height (considering any physics overrides) + 1. Anything else gives something that feels weirdly inconsistent. I don't want to need to rely on mods to try and get round this. It's simply not going to be fast/responsive enough.

There are other minor issues as well that will keep putting me off new movement, such as reduced distance over the edge, losing the ability to jump while sneaking and so on.

I've made quite a few attempts since the change to just get used to it, but ultimately I simply end up finding the game not fun anymore. And when you don't find a game fun anymore, there's not much reason left to carry on with the game.

**Comment by benrob0329** [comment]
Some ideas:

* Being able to pull yourself onto a ledge (not sneak style, more parkour style)
* (Smooth!) Sprint jumping
* Rolling (to reduce fall damage from reasonable heights)
* Some sort of climbing, again not classic sneak style but something more natural

Basically, take the movement limitations that the old bugs had, and make ways of not having such restricted movement. But don't try to blindly replicate the old behavior, instead make something much better than can be controlled by the game (perhaps you need to break your legs a few times before you successfully roll, or sometimes lose your grip/can't grip at all when climbing).

Exploring could go from "OK I need some dirt to scaffold across this cavern" to "OK, I need a run-up and to catch that ledge, with a slight possibility of falling into the cavern". It would help make such things more exciting.

**Comment by paramat** [comment]
4w wrote:
> Remove the bug code and bug simulation code without replacement. Everyone who wants to have it back should use a mod for that.

A mod can't create the same behaviour, it's clientside movement code so not remotely moddable.
4w wrote:
> Of course people like you are butthurt about fixed bugs (hence the downvote I guess). This does not make bugs better in any way. Just don't rely on bugs for your functionality/gameplay.

If you were present during the sneak controversy you know your suggestion is not acceptable and is a provocation, please don't stir up the arguments and bad feelings from the controversy. My suggestion is not controversial, is labelled low priority and keeps the players that like old sneak behaviours happy, it is not related to the arguments of the past.

I am discussing this only on the condition that we do not drag up the old controversy. Warning: i may delete comments that do that.
4w wrote:
> Politeness is irrelevant in software development

Being reasonably civil is a dev rule in MT, you're often close to pushing unpleasantness too far, please be less abrasive. The only moderation power we have a a complete ban from the repo, we don't want to do that.
4w wrote:
> A bug is not a feature and should not be seen as one

Some bugs become well liked features that become essential to some, this is obviously the case with sneak.
4w wrote:
> If constant bug using leads to a certain behavior by a lot of users then the bug should be fixed and the effects of the bug should be properly re-implemented as optional (-> maximum configurability) feature.

If you were around during the controversy you know this was done, was a mistake, the simulations were not exact, and it caused a huge amount of grief.

Ezhh is completely correct about the failings of the new sneak jump, it is not a replacement that will keep players happy.

**Comment by 4w** [comment]
> A mod can't create the same behaviour, it's clientside movement code so not remotely moddable.

Then the modding API is incomplete. What about CSM that was explictly designed to mod the client via server-sent client-mods.

> Some bugs become well liked features [...]

You spelled "Some bugs result in well like behavior" wrong.

> [...] that become essential to some

As said and implied earlier: If your gameplay relies on results of unfixed bugs it is entirely your fault if something breaks after the bug was fixed.

> If you were around during the controversy

I was, and I was super happy when this long-term bug was finally fixed and I was super sad when I learned that the bug was kept.

**Comment by jastevenson303** [comment]
I haven't played since bindable mouse button PR conflicts and I was unsable
to resolve.  Without sneak jumping - an important feature - it's doubtful
I'd ever play again.

**Comment by benrob0329** [comment]
>Without sneak jumping - an important feature - it's doubtful I'd ever play again.

These comments make me sad, about as sad as

>As said and implied earlier: If your gameplay relies on results of unfixed bugs it is entirely your fault if something breaks after the bug was fixed.


Sneak jump is not going away, classic sneak code is staying as new sneak isn't quite polished yet and the new behavior is not a replacement for the old. This discussion is not about keeping old sneak, its about whether or not to completely redesign movement functionality in the future without trying to copy old behavior that is really hard to do in a good way.

That being said, it is immensely important to make gameplay fun. Removing the ability to move in ways players have learned to love can only be done **if a much better method is created.** Currently, new sneak is not very much fun to use, and we don't have any new functionality to help augment movement.

**Comment by Megaf** [comment]
I didn't read the previous comments because well... Not worth it.

My opinion is that the old sneak should have not have been removed in the first place, it was part of the gameplay, people have been using it for several years. And on my server, with the help of people in #Minetest-Hub we reimplemented it and my few players are happy with it, including myself.

I would be thrilled in having sneak back to the game.

**Comment by paramat** [comment]
4w wrote:
> Then the modding API is incomplete. What about CSM that was explictly designed to mod the client via server-sent client-mods.

There are many things that mods including CSM mods won't be able to do. I doubt that CSM could exactly simulate the old sneak behaviour.

jastevenson303 to be clear, don't worry, old sneak behaviours are not going to be removed.

**Comment by paramat** [comment]
> One thing you might want to get some opinions on is whether those using new movement on some servers also want to preserve sneak ladder

> What also might be interesting are a few poll results that show what the players actually use, so this issue can be resolved easier by deciding whether it's worth removing options/code or not.

Hmm actually this is the reason why i think it's best the simulations are removed as soon as possible before players start to rely on them.
Krock and i spent very many hours perfecting and testing the sneak jump / ladder simulations, they were very difficult to tune and are sensitive to changes in movement code. I spent more hours testing that than any other PR.

If some players get used to and rely on the new simulations then we have added a second set of specific movement behaviours that will need to be preserved, they are highly likely to break in the future as we work on movement code due to their sensitive nature, meaning we run the risk of repeating the past controversy again.

For this reason alone it is well worth removing the simulations immediately to make movement code simpler, more robust and reducing future maintenance and controversies. I wouldn't even bother finding out if some like them, the perfected simulations were only released very recently in stable so there's little pain in removing them, however the longer we leave it the more pain it will be.
Note that the simulations are off by default so there's even less chance of players getting used to them.
Also this is 0.5, the best time to break stuff.

**Comment by paramat** [comment]
Bump for more consideration, please see my previous comment for my current opinion.
I can't see any reason to keep simulations of sneak jump and sneak ladder, and they are likely to cause problems for the future.

**Comment by paramat** [comment]
I should clarify again: This issue does not relate to the past controversy, the old move code and old sneak behaviour will remain as an option. Anyone who tries to stir that up (as above) misunderstands or is offtopic.

**Comment by benrob0329** [comment]
I still agree with removing the imitation sneak, but would prefer that new sneak have more features added (perhaps crouching/probing would be a nice addition)

**Comment by paramat** [comment]
So, please can we remove these simulations? They are a danger, are little used, and cause extra problems for movement code development. I'll make the PR if pre-approved.

**Comment by paramat** [comment]
Doesn't seem popular, closing.

**Comment by paramat** [comment]
Disapproved so will close.

However, it is very important that we state that we will not necessarily preserve any non-intentional movement behaviour, such as new unintentional sneak-jumps (not the intentional simulations of old sneak jump and sneak ladder) otherwise we will end up repeating the controversy multiple times, or otherwise be unable to improve or fix movement behaviour for fear of controversy. So i will make a news forum post that states this.

The old move sneak-jump and sneak-ladder were usable for many years and had become much liked features, and the core devs did not at any point warn against depending on them (which is fine), so these became worth preserving. But we need to avoid this happening in the future.

**Comment by paramat** [comment]
Reopen, was not actually disapproved in this thread.
Also, #9343 has made this relevant, and what i predicted has happened: Improvements to new movement code are being proposed, but are incompatible with sneak jump/glitch.

In addition i now propose we also remove sneak glitch from new movement code. So that all these unintended-but-valued sneak-related behaviours are confined to the old movement code, freeing up the development of new movement code.

**Comment by TheTermos** [comment]
Looks like I might have triggerred reopening of this issue. Let's hope this time the purpose of it being reopened isn't so everybody can say nothing.

I wonder if @SmallJoker  (the last one left with his thumb hanging) could explain what's so wonderful about having both sneak glitch in the old move and its emulation option in the new move, where it hinders development and makes fixing bugs unnecessarily difficult, if possible at all. Also how to reconcile the notion of Minetest trying to become a game engine with having extremely specific and peculiar gameplay mechanics that require a degree in Minetest history to understand.

I also wonder if the new guys, @pyrollo  and @v-rob  have an opinion about this. They are probably not to blame for this mess, so they might have an unbiased view.

Also @rubenwardy, he's been sending mixed signals about this. I wonder if he can make up his mind and  say what he means, you know, with words.

Anyway, if anyone were interested in my humble opinion about this, I'd say: remove the soonest.

**Comment by SmallJoker** [comment]
Short version:
I believe this whole story was caused by a bad start in #5327. The PR broke the sneak glitch (ladders) which some people have considered as a gameplay feature. This main concern was solved a long time ago.
Also limiting the player extent to the box corners (to not fall down on sneak release) was apparently an issue because certain locations are more difficult to reach, or to jump over gaps.

The new code was introduced in mid 2017 and I think it's about time to remove the old stuff, although it did (AFAIK) not require any maintenance work so far.

**Comment by sfan5** [comment]
> where it hinders development and makes fixing bugs unnecessarily difficult, if possible at all.

Is there a concrete example of something that could not, can't or will not be able to fixed due to these "features" (if one wants to call them that)?

Or are you going to hide behind your weirdly confrontational attitude and just let that claim stand on thin air?
Because honestly (forgive the offtopic rant) this has been pissing me off for a while now. When I read your many of your comments here I instinctively don't even **want** to reply. They give off a sense of "My opinion is always right", "You must all be stupid that you cannot see the correct way" and "Nothing works out anyway because of you guys". Your points are not wrong, but it would help immensely if you wrote them out in a way that was less toxic.

**Comment by v-rob** [comment]
I really like the sneak glitch, but when I really want to play with it, I take a trip down memory lane and play with it in 0.3.1, not re-enable it in the current version.  If the sneak glitch really is a problem for our code, I probably wouldn't oppose its removal.  If it isn't a problem and likely never will be, I would say that it might as well stay.

**Comment by TheTermos** [comment]
> When I read your many of your comments here I instinctively don't even **want** to reply. They give off a sense of "My opinion is always right"

I haven't seen you take part in this discussion before, nor have you been mentioned here, so I guess your reaction must have been triggered elsewhere.
Look, feelings and sensations are not my forte, If you say you're getting them then you do, no way for me to argue. Yes I've been critical about certain devs' actions, including yours. I know nobody likes being criticized, now there's the issue to solve.

> Is there a concrete example of something that could not, can't or will not be able to fixed due to these "features" (if one wants to call them that)?

Obviously,
That's why this issue had been reopened last year. You might start with paramat's post above which contains links to relevant discussions, nothing new has happened since then.

Also a reminder https://github.com/minetest/minetest/pull/5327
> I feel like I'm repeating myself here: Sneak ladders rely on sneaking being so broken, in the process of fixing any of the other bugs the sneak ladder breaks, that's just how it is. I did not specifically want to remove it.

This is pretty obvious stuff and this opinion still holds, localplayer is still buggy and very badly coded. Have you changed your mind since then?

**Comment by sfan5** [comment]
> I haven't seen you take part in this discussion before, nor have you been mentioned here, so I guess your reaction must have been triggered elsewhere.
Look, feelings and sensations are not my forte, If you say you're getting them then you do, no way for me to argue. Yes I've been critical about certain devs' actions, including yours. I know nobody likes being criticized, now there's the issue to solve.

By "here" I meant Minetest's issue tracker entirely. And no, criticism isn't the problem, it's how one goes about it.

> #9343

Right, I can see that some sneak-related fixes were not included because they make the sneak glitch not work.

It makes sense that keeping the code in a broken state so that the sneak glitch happens "naturally" is not a good idea.
But I'm wondering does re-adding the sneak glitch _artificially_ get in the way of improvements?

For reference, the defining feature of the glitch is being able to sneak "on" the lower node L in such a construction.
<pre>
[X]          ↑
             |
[L]          | Y
</pre>
In addition, you need to able to jump from that state to stand on node X (how does that even work?).

> Have you changed your mind since then?

No, this is still true. I wasn't precise on how I think the glitch could be kept initially (see above).

**Comment by TheTermos** [comment]
> It makes sense that keeping the code in a broken state so that the sneak glitch happens "naturally" is not a good idea.
> But I'm wondering does re-adding the sneak glitch _artificially_ get in the way of improvements?

So we seem to agree afterall.

Thing is the glitch still happens "naturally". 
I've got nothing against proper, intentional features like climbing (but only on the assumption that player stays clientside, which is itself not a great thing, that's another discussion though)

> For reference, the defining feature of the glitch is being able to sneak "on" the lower node L in such a construction.

As far as I remember, this is possible for 2 reasons:
- buggy geometry tests that fail to take into account relevant details of the environment.
- player displacement bypasses collision detection (set pos explicitly), so its box freely intersects with nodes.

My solution worked on the assumption that player should never clip into nodes, and rather than bypassing collision detection used it to the advantage, that also made it simpler.

Now how to legitimately get on the L node with such approach: player collisionbox is now configurable, isn't it? Crouch animation is also possible I guess. Chances are this already can be achieved via mod and the engine doesn't need to bother.

> In addition, you need to able to jump from that state to stand on node X (how does that even work?).

I think the original sneak ladder worked differently, there was that checkered pattern, so the mechanics was similar to that of getting on the L node - as soon as you climbed L the one next to you became the new L.
If I understand that thread correctly, the ability to sneak on the node X directly above has been introduced by your PR, so hopefully you know better than me how that works ;)

**Comment by sfan5** [comment]
> I think the original sneak ladder worked differently

Yes, I tried to minimize the description but looks the second requirement shouldn't be there.

> the ability to sneak on the node X directly above has been introduced by your PR,

Entirely possible...

> so hopefully you know better than me how that works ;)

...but I do not remember my code from 4 years ago, so no.

**Comment by TheTermos** [comment]
I've looked it up, starts here:
https://github.com/minetest/minetest/blob/b5eda416cea3157ae3590fb7d229cd2cd17c3bf9/src/client/localplayer.cpp#L152
It's been added in its current form by another dev, not you, but your version reportedly had the same outcome, so possibly similar flaws. 

It's done the dumbest way possible: if n+2 walkable and n+3 nonwalkable then move on top of n+2.
Just looking at the code it's immediately apparent that this would allow player to glitch through a layer of solid nodes and in fact, I've been able to reproduce it.

So it seems that attempts of preserving the glitch have been made without even understanding how the original glitch happened. In effect new bugs have been introduced, potentially for players to fall in love with and demand preserving. Looks much like snowball effect to me.


This is not an isolated instance, making changes to the code on a hunch, without understanding how it works is commonplace here, a rule rather than an exception, done by random contributors and core devs alike.
You probably understand this doesn't look good at all, and I could still go on, so back to your complaints about my style - some messages need to be conveyed to the dev team that some of them might find uncomfortable, and for good reasons.
I wish this project well, and I'd like to help it within my limited resources - the project, I don't necessarily care much about members' self esteem. I know that's not a nice thing to say but I'm being honest here.
But you should know that I'm trying not to loose my patience too much, and that I'm still holding back more than I let on, so I hope we can agree there's no need to waste any more time discussing my style.

**Comment by sfan5** [comment]
> It's done the dumbest way possible: if n+2 walkable and n+3 nonwalkable then move on top of n+2.

What do you mean by "move on top of n+2"?
I can see that the code allows movement and jumps when such a 'sneak ladder' is detected. Is that what you meant?
It also [forces the position however far up it needs to be](https://github.com/minetest/minetest/blob/b5eda416cea3157ae3590fb7d229cd2cd17c3bf9/src/client/localplayer.cpp#L381-L384) (though that's not exclusive to the sneak glitch).

> so I hope we can agree there's no need to waste any more time discussing my style.

...and your suggestion to make it better is?

**Comment by TheTermos** [comment]
> What do you mean by "move on top of n+2"?
> I can see that the code allows movement and jumps when such a 'sneak ladder' is detected. Is that what you meant?

Yes, maybe too much of a shortcut, that particular part doesn't move it directly, it just adds the ability to jump in midair to the ability to move inside nodes, which eventually make that move possible.

> ...and your suggestion to make it better is?

I'm not sure which 'it' you mean here.

**Comment by sfan5** [comment]
> I'm not sure which 'it' you mean here.

The code in localplayer that brings the sneak glitch back.
Since you said it allows glitching through a solid layer of nodes (interested to hear the exact setup for that too), the code should be improved and that bug fixed.

**Comment by TheTermos** [comment]
![sladder](https://user-images.githubusercontent.com/55103816/109354355-f74a0300-787d-11eb-9d9b-8e2a41ccd4bd.gif)

You know what I think of glitches, a good one is a dead one, it's best improved by being removed.

As to localplayer features in general:
- best solution: move player server side and let mods handle (plus that solves a ton of other issues)
- worse: if player stays client side, a climb feature could work based on presence of exposed Y+ faces adjacent to player's collisionbox

**Comment by Wuzzy2** [comment]
I was a long-time enemy of the sneak glitch because it came with a lot of related problems, and not all games would want to allow sneak elevators.

I was a big fan of the bugfixes related to sneaking, unsurprisingly. Later, the “simulations”, as paramat calls them, came. I actually like that idea a lot, because it means the sneak glitch is optional. This is great. I totally get that sneak elevators are useful and have a huge fan base. However, I also think they are massively OP and they are clearly not for all games. Optionality is very important.

As I see it, the main reason for the Sneak Glitch's popularity is that sneak elevators work, this is the one reason why people want it. I am all for having sneak elevator support, optional. With the expressed use case of making sneak elevators work as an intended *feature*, but without ANY unintended side-effects (aka bugs).

I don't really care about Old Move, but I think it needs much better documentation, because it affects many things at once (including the horrible fall damage evasion, sadly); I'm afraid people will use it without really understanding it.

It's a good idea to have the `sneak_glitch` player physic, and it supports sneak elevators okay. Therefore, I strongly support **keeping** the `sneak_glitch` option.
Sadly, you can still glitch through the ceiling with the Sneak Glitch (although its quite tricky), see #11081. This is a bug which should be fixed.

There is one other aspect in Old Move: You jump slightly higher if you hold the Sneak button. I think it might be a good idea to turn this single aspect into a physics modifier in its own right (maybe with a customizable jump height as well). Although  personally do not care at all about this, other players definitely do.

To summarize:
* Don't touch Old Move (for historic reasons)
* Keep the `sneak_glitch` player physic
* Test the `sneak_glitch` player physic to make sure it does not come with ANY unintended side-effects
* Fix glitching through the ceiling (#11081)
* Consider adding a new player physic for sneak jumping (increased jump height while sneaking, idea from Old Move)
* Consider adding a `sneak_glitch` setting for Minetest to give players a quick and easy way to re-enable sneak elevators

**Comment by Wuzzy2** [comment]
> Consider adding a sneak_glitch setting for Minetest to give players a quick and easy way to re-enable sneak elevators

I meant “Minetest Game”. Whoops!
